### ISSUE TITLE
`at-rule-empty-line-before` was removed from v9 but it should be returned

### ISSUE THREAD

**Issue by boris-petrov** [prettier/stylelint-config-prettier#128]
I updated from 8.0.2 to 9.0.2 and I got a few `at-rule-empty-line-before` warnings. Most of them I can fix but one is conflicting with Prettier:

```scss
@if ($x == false) {
} @else {
}
```

This gives the warning on the `@else` line. However, putting a new-line there is 1) ugly and 2) removed by Prettier. So this rule should be disabled in `stylelint-config-prettier`.

**Comment by and1-lt** [comment]
With this rule being removed, prettier doesn't complain about missing lines between multiline at blocks, example - there's no errors on given code, though it really doesn't look well formatted:

```
padding: 20px;
@include theme.mediaTabletLandscapeUp {
   width: 500px;
}
@each $radiusKey, $radiusValue in $_radiusConfig {
  &.#{$radiusKey}Radius {
    border-radius: $radiusValue;
  }
}
@each $colorVariantKey, $colorVariant in $_colorVariantConfig {
  &.#{$colorVariantKey} {
    @if $colorVariantKey != 'light' {
      color: theme.$colorWhite;
    }

    background-color: $colorVariant;
  }
}
```

So, once again, should `at-rule-empty-line-before` really be disabled?
If the only conflict is with `@else`, then the `at-rule-empty-line-before` has an option to ignore it - `ignoreAtRules: ['else']`, whch is also enabled on `stylelint-config-standard-scss`

**Comment by shannonmoeller** [comment]
Hi! Thanks for the comment. This one is tricky. I definitely see the utility of this rule and how disabling it can be annoying. But, given this rule causes a conflict with prettier (meaning running `stylelint` first and `prettier` second results in different output than `prettier` first and `stylelint` second) I think this rule should remain disabled by default. It's still possible to re-enable it for projects that desire it by readding the rule after including this config which feels like it might be the right compromise?

Manually per-project:

```json
{
  "extends": [
    "stylelint-config-standard-scss",
    "stylelint-config-prettier"
  ],
  "rules": {
    "at-rule-empty-line-before": [{ ignoreAtRules: ['else'] }]
  }
}
```

Or by creating a new config package that reenables what you want:

```json
{
  "extends": [
    "stylelint-config-standard-scss",
    "stylelint-config-prettier",
    "stylelint-config-whatever-you-want-to-call-it"
  ]
}
```

I'm happy to continue the discussion if you want to open a new issue.
